### PR TITLE
Added useClickAnyWhere

### PR DESCRIPTION
**Pull request by sonjoydatta**
Hi @juliencrn, 
I just added a simple hook called "useClickAnyWhere". 
Can you review it?

Thanks.

### CODE REVIEW

File: packages/usehooks-ts/src/useClickAnyWhere/useClickAnyWhere.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { useEffect } from 'react'
+
+type AnyEvent = MouseEvent | TouchEvent
+type Handler = (event: AnyEvent) => void
```
Comment (juliencrn) [inline packages/usehooks-ts/src/useClickAnyWhere/useClickAnyWhere.ts:4]: Can be simplified to 

```suggestion
type Handler = (event: MouseEvent) => void
```

Indeed, the `TouchEvent` is keyboard related and not called in the hook.

File: packages/usehooks-ts/src/useClickAnyWhere/useClickAnyWhere.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+function useClickAnyWhere(handler: Handler) {
+  useEffect(() => {
+    const callback = (event: AnyEvent) => handler(event)
+
+    document.addEventListener('click', callback)
+    return () => document.removeEventListener('click', callback)
```
Comment (juliencrn) [inline packages/usehooks-ts/src/useClickAnyWhere/useClickAnyWhere.ts:11]: Here, two things:
- We can simplify the hook using the `useEventListener`.
- The Typescript types check doesn't pass, we have to type it manually. 

```suggestion
    useEventListener('click', event => {
      handler(event as unknown as MouseEvent)
    })
```

**Comment by juliencrn** [comment]
Hello @sonjoydatta!

Thank you for your nice work, and good idea, we will add it 👍

There are some points to see:
- The `develop` branch is a little outdated, my bad, could be better to rebase on `master`.
- Do you also want to add the description to display it on the site? See the `npm run plop` command ;)

PS: I discover an issue in another event based hook while doing this code review, cool!

**Review by juliencrn — CHANGES_REQUESTED** [review]

File: packages/usehooks-ts/src/useClickAnyWhere/useClickAnyWhere.ts (hunk 2 above)
Comment (sonjoydatta) [reply packages/usehooks-ts/src/useClickAnyWhere/useClickAnyWhere.ts:11]: Good idea!

**Comment by juliencrn** [comment]
Branch moved to #64

**Pull request closed without merging**